src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1557,11 +1557,17 @@ private void SetPeriodicEvaluation(WorkspaceModel ws)
               homeWorkspace.RunSettings = new RunSettings(runType, runPeriod);
             }
 
-            RegisterHomeWorkspace(homeWorkspace);
-
             CustomNodeWorkspaceModel customNodeWorkspace = workspace as CustomNodeWorkspaceModel;
             if (customNodeWorkspace != null)
-              customNodeWorkspace.IsVisibleInDynamoLibrary = dynamoPreferences.IsVisibleInDynamoLibrary;
+            {
+                customNodeWorkspace.IsVisibleInDynamoLibrary = dynamoPreferences.IsVisibleInDynamoLibrary;
+            }
+
+            // Only RegisterHomeWorkspace if not a CustomNodeWorkspace
+            else
+            {
+                RegisterHomeWorkspace(homeWorkspace);
```

Comment:
Can you put this function call to the last line in `if (homeWorkspace != null)`? I feel that's a better place since we already pretty sure it would be a `homeworkspace` there

Reply:
ah yes, good call